### ISSUE TITLE
Update Readme to include a section explaining what is Discord.

### ISSUE THREAD

**Issue by blam1992** [Cog-Creators/Red-DiscordBot#6183]
### What component of Red (cog, command, API) would you like to see improvements on?

Category: Docs - Readme

### Describe the enhancement you're suggesting.

Since Discord is required in order to use Red Discord Bot. I believe it would be beneficial to new users/contributors who wants to learn more about this project, despite not knowing what Discord is. This section will include a short description of what Discord is and provide links to download the software (depending on the user's operating system).

**Comment by Jackenmen** [comment]
I like this idea, it seems helpful for someone to be able to look this project up and be able to actually figure out what it's for. Selfishly, I could just link people the project's GitHub site instead of trying to explain myself.

Aa for the specifics, we can have a download link but I wouldn't want to have to specify it separately for each OS as you suggest - let's just have one URL to discord.com and let it handle the OS detection as it already does.

I don't know if this is of any use but this is the description I used before:
> Red is a modular self-hosted Discord bot written in Python that is extremely customizable (and extendable through plugins) and has gained a large community of content creators. Discord itself was made for gamers but has developed with the interaction of communities to create a fully featured chat/voice client, being used for both everyday conversations and work-related conversations. Discord has a wonderfully documented API that encourages developers and geeks alike to explore their creativity and create programs, tools, and services to enhance their experience.

I would suggest rewording that last part a bit though since it's speaking a bit too highly of Discord ;)

**Comment by blam1992** [comment]
Thanks for the quick response, I will avoid having separate download links for each OS. Below a short description of Discord:

> Discord is a voice, video, and text chat application that allows users to communicate with their friends and communities. It is one of the most popular communication services for gamers, and it is free to use. Discord utilizes servers that contain chat rooms and voice channels that can be accessed through invite links.

Let me know what you think.

**Comment by Flame442** [comment]
> Red is a modular chat bot written in Python that is extremely customizable (and extendable through plugins) and has gained a large community of content creators. Red is a self-hosted bot – meaning you will need to host and maintain your own instance. You can turn Red into an admin bot, music bot, trivia bot, new best friend or all of these together!
> 
> Red is built to be used in [Discord](https://discord.com/), a voice, video, and text chat application that allows users to communicate with their friends and communities. Discord utilizes servers that contain chat rooms and voice channels that can be accessed through invite links. 

Something closer to this probably. Remember, this is the very beginning of the readme that is supposed to be an explanation about *Red*, so it still needs to contain details about what Red itself is. I think a brief 2 sentence explanation of discord, and linking to their site with more information, is more than enough.
